### ISSUE TITLE
package:media_kit

### ISSUE THREAD

**Issue by alexmercerind** [alexmercerind/dart_vlc#379]
### TL;DR

Please consider replacing package:dart_vlc with [package:media_kit](https://github.com/media-kit/media-kit) in your Flutter projects. It should require minimal changes.

[package:media_kit](https://github.com/media-kit/media-kit) brings major advancements. Few are:
- Support for all platforms _i.e._ Android, iOS, macOS, Windows, GNU/Linux & Web.
- Hardware acceleration _i.e._ GPU usage, allowing playback of multiple FHD, 4K & 8K videos.
- Support for both video & audio applications.
- Unit-tests.
- Smaller size.
- Built-in video controls.
- Improved API.
- Improved stability.
- Improved performance.
- Less restrictive license: MIT.

[Contributions are welcomed to package:media_kit](https://github.com/media-kit/media-kit).

Let's have a reliable video & audio playback library for Flutter, which meets community's demand.

<hr>

### Why

From a personal standpoint, working on package:dart_vlc has been a good learning experience. I started working on this project when I was around 18 years old. This helped me learn in detail about C/C++, low-level APIs (such as libVLC & Flutter platform channels), native interop, shared libraries etc. With that being said, I strongly believe that code-quality & stability has been highly compromised due to my lack of experience & knowledge at that time.

On the other hand, package:dart_vlc is very inefficient as it doesn't utilize GPU for video playback, causing excessive CPU usage & reduced battery life. There are other limitations in libVLC, which were becoming a blocker for the overall experience of decent video playback. Lastly, the bundle size is also very large.

### Future

At the moment, I'm investing my time in the further development, advancement & maintenance of  [package:media_kit](https://github.com/media-kit/media-kit). In future, if the requirement of a video package that specifically uses libVLC is felt, I will consider reviving this project & re-write things from scratch.

For the note, [VLC Media Player](https://www.videolan.org/vlc/) is an extremely good piece of software capable of playing ~most~ all multimedia files, I use it nearly everyday since my childhood. However, libVLC isn't exactly seeming like the perfect-fit here.

Thanks!
